services/forms/repo_form.go

```diff
@@ -689,7 +690,7 @@ func (f *NewWikiForm) Validate(req *http.Request, errs binding.Errors) binding.E
 // EditRepoFileForm form for changing repository file
 type EditRepoFileForm struct {
 	TreePath      string `binding:"Required;MaxSize(500)"`
-	Content       string
+	Content       optional.Option[string]
```

Comment:
I don't know how `optional.Option[string]` is handled here.

Reply:
It's a type alias to `[]string`, the parsing is done in [go-chi/binding](https://gitea.com/go-chi/binding/src/commit/39a851e106edb79a30887aff4df957e2465ecbbd/binding.go#L576). If no form value is given, the field will just use the default value, which is nil which is `Option.Has == false`, else the form string will be converted into a []string with the form string as it's only value, as far as I can tell. 
We could also use a hidden input like `<input name="has_content" type="hidden" value="something">` instead.